src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,111 @@
+                        itemList.markTask(index);
+                        input = sc.nextLine();
+                        break;
+                    // unmark items
+                    case "unmark":
+                        index = Parser.getIndex(input);
+                        itemList.unmarkTask(index);
+                        input = sc.nextLine();
+                        break;
+                    case "todo":
+                        if (Parser.isValidDescription(input)) {
+                            Task toAdd = new ToDo(Parser.getDescription(input));
+                            itemList.addTask(toAdd);
+                            input = sc.nextLine();
+                        } else {
+                            throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+                        }
+                        break;
+                    case "deadline":
+                        if (!Parser.isValidDescription(input)) {
+                            throw new DukeException("OOPS!!! The description of a duke.Deadline cannot be empty.");
+                        } else if (!Parser.isValidDateTime(input)) {
+                            throw new DukeException("OOPS!!! The time and date of the duke.Deadline cannot be empty.");
+                        } else {
+                            //eg. by 2019-10-03 18:00
+                            Task toAdd = new Deadline(Parser.getDescription(input),
+                                    Parser.getDate(input),
+                                    Parser.getFrom(input));
+                            itemList.addTask(toAdd);
+                            input = sc.nextLine();
+                        }
+                        break;
+                    case "event":
+                        if (!Parser.isValidDescription(input)) {
+                            throw new DukeException("OOPS!!! The description of a duke.Event cannot be empty.");
+                        } else if (!Parser.isValidDateTime(input)) {
+                            throw new DukeException("OOPS!!! The time and date of the duke.Event cannot be empty.");
+                        } else {
+                            Task toAdd = new Event(Parser.getDescription(input),
+                                    Parser.getDate(input),
+                                    Parser.getFrom(input),
+                                    Parser.getTo(input));
+                            itemList.addTask(toAdd);
+                            input = sc.nextLine();
+                        }
+                        break;
+                    case "find":
+                        itemList.findTask(Parser.getDescription(input));
+                        input = sc.nextLine();
+                        break;
+                    // unrecognised commands
+                    default:
+                        throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
```

Comment:
No indentation for `case` clauses according to the module's [Java Coding Standard](https://se-education.org/guides/conventions/java/intermediate.html).

Reply:
Ok thanks for the advice